src/transformers/models/grounding_dino/convert_grounding_dino_to_hf.py

```diff
@@ -0,0 +1,423 @@
+    return image
+
+
+def preprocess_caption(caption: str) -> str:
+    result = caption.lower().strip()
+    if result.endswith("."):
+        return result
+    return result + "."
+
+
+@torch.no_grad()
+def convert_grounding_dino_checkpoint(args):
+    model_name = args.model_name
+    pytorch_dump_folder_path = args.pytorch_dump_folder_path
+    push_to_hub = args.push_to_hub
+
+    checkpoint_mapping = {
+        "grounding-dino-tiny": "https://huggingface.co/ShilongLiu/GroundingDINO/resolve/main/groundingdino_swint_ogc.pth",
+        "grounding-dino-base": "https://huggingface.co/ShilongLiu/GroundingDINO/resolve/main/groundingdino_swinb_cogcoor.pth",
+    }
+    # Define default GroundingDINO configuation
+    config = get_grounding_dino_config(model_name)
+
+    # Load original checkpoint
+    checkpoint_url = checkpoint_mapping[model_name]
+    original_state_dict = torch.hub.load_state_dict_from_url(checkpoint_url, map_location="cpu")["model"]
+    original_state_dict = {k.replace("module.", ""): v for k, v in original_state_dict.items()}
+
+    # Rename keys
+    new_state_dict = original_state_dict.copy()
+    rename_keys = create_rename_keys(original_state_dict, config)
+
+    for src, dest in rename_keys:
+        rename_key(new_state_dict, src, dest)
+    read_in_q_k_v(new_state_dict, config)
+
+    # Load HF model
+    model = GroundingDINOForObjectDetection(config)
+    model.eval()
+    missing_keys, unexpected_keys = model.load_state_dict(new_state_dict, strict=False)
+    print("Missing keys:", missing_keys)
+    print("Unexpected keys:", unexpected_keys)
+
+    # Load and process test image
+    image = prepare_img()
+    transforms = T.Compose([T.Resize(size=800, max_size=1333), T.ToTensor(), T.Normalize(IMAGENET_MEAN, IMAGENET_STD)])
+    original_pixel_values = transforms(image).unsqueeze(0)
+
+    image_processor = GroundingDINOImageProcessor()
+    tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
+    processor = GroundingDINOProcessor(image_processor=image_processor, tokenizer=tokenizer)
+
+    text = "a cat"
+    inputs = processor(images=image, text=preprocess_caption(text), return_tensors="pt")
+
+    assert torch.allclose(original_pixel_values, inputs.pixel_values, atol=1e-4)
+
+    # Running forward
+    with torch.no_grad():
+        outputs = model(**inputs)
+
+    print("First values of logits:", outputs.logits[0, :3, :3])
+    print("First values of boxes:", outputs.pred_boxes[0, :3, :3])
+
+    # verify outputs
+    expected_boxes = torch.tensor([[0.7674, 0.4136, 0.4572], [0.2566, 0.5463, 0.4760], [0.2585, 0.5442, 0.4641]])
+    expected_logits = torch.tensor(
+        [[-4.8915, -0.1900, -0.2161], [-4.9658, -0.3716, -0.3948], [-5.9596, -3.3763, -3.3103]]
+    )
+    assert torch.allclose(outputs.pred_boxes[0, :3, :3], expected_boxes, atol=1e-4)
+    assert torch.allclose(outputs.logits[0, :3, :3], expected_logits, atol=1e-3)
+    print("Looks ok!")
```

Comment:
tests should go in the testing file no the conversion script. Or at least make them optional

Reply:
https://github.com/huggingface/transformers/pull/26087/commits/afb26499a15f4d7ea68d271697f5348131811971